### ISSUE TITLE
jitsi meet view error architecture

### ISSUE THREAD

**Issue by jalalzeadeh** [jitsi/jitsi-meet#3436]
When i use JitsiMeetView in my UIViewController i get error:
ignoring file ...../JitsiMeet.framework/JitsiMeet, file was built for arm64 which is not the architecture being linked (x86_64): ..../JitsiMeet.framework/JitsiMeet
Undefined symbols for architecture x86_64:
  "_OBJC_CLASS_$_JitsiMeetView", referenced from:
      objc-class-ref in ViewController.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by ranggamp** [comment]
Same as mine, but before I using xcode 10 this was just fine, I was updated Xcode to version 10 which is supported for iOS 12, I have to re-build Jitsi and WebRTC Framework, following manual but no luck.
Try this https://stackoverflow.com/questions/45633435/archive-validation-error-unsupported-architecture-invalid-segment-alignment-e/45659911#45659911
also with no luck.

**Comment by saghul** [comment]
@jalalzeadeh You seem to have built binaries for a device and are trying to run it on a simulator.

@ranggamp I don't know what your problem is, can you rephrase? You shouldn't need to recompile the WebRTC framework at all.

**Comment by ranggamp** [comment]
Thanks @saghul, but since it's compile using swift 4.1 then I couldn't use this framework on xcode 10 which is suggested by xcode itself.

**Comment by saghul** [comment]
Hum, I'm not sure how that can happen. The WebRTC framework doesn't have any Swift in it.

**Comment by jorgearreola** [comment]
i have the same issue!
today update to Xcode10, and I need to rebuild jitsi framework but ut something changes that no longer generates the files correctly and the architecture errors appear...

any idea?

**Comment by ranggamp** [comment]
I think I might have a solution for this. Copy this JitsiMeet binary to JitsiMeet.framework
[JitsiMeet_For_Device.zip](https://github.com/jitsi/jitsi-meet/files/2440426/JitsiMeet_For_Device.zip)
[JitsiMeet_For_Simulator.zip](https://github.com/jitsi/jitsi-meet/files/2440427/JitsiMeet_For_Simulator.zip)

**Comment by saghul** [comment]
Folks, can anyone here write down *the exact steps* you are following? I've been using Xcode 10 since the betas and have never run into this.

**Comment by sanalpencere** [comment]
I have same situation too. @ranggamp thanks for binaries. At least I can build application and test it. But time to time I get react-native errors. 

I am trying to solve problem. If anyone does please keep us updated.

**Comment by jorgearreola** [comment]
Hello everyone! After trying and trying, I share how I could solve it.

My problem: When I migrated to XCode10, I got a JitsiMeet.framework error because it was compiled in another version of my project.

Solution: When I opened the project (ios / jitsi-meet.xcworkspace) and compiled it in Xcode10 I had no problems and ran the demo example (example-pip-app). However, when doing the final release process to generate the .framework in command line (xcodebuild -workspace ios / jitsi-meet.xcworkspace -scheme JitsiMeet -destination = 'generic / platform = iOS' -configuration Release archive), it's works fine, but when integrating JitsiMeet.framework release to my final project, i had architecture errors (it only compiled for x86_64 and i386 architectures), when in the previous versions of XCode I perfectly generated the JitsiMeet.framework in arm64 and armv7 architectures. I managed to do it directly from XCode: I changed my Debug mode to Release, I ran the example ios / jitsi-meet.xcworkspace and when generating the JitsiMeet.framework in the project reference, I went to the location and took the JitsiMeet.framework release path, I copy it to my final project and now if I compile correctly without errors, I validate the architectures and in fact the ones I needed were already compiled: arm64 and armv7

So: If compiling by command line does not generate the JitsiMeet.framework (architectures) correctly, but compiling from XCode does it correctly!!!

I hope my solution can help someone else.

**Comment by Frindecent** [comment]
HI , After long trying and i also posted this as a bug initially when i couldnt understand about this. I found an easy workaround. Please check this and let me know if that helps

https://github.com/jitsi/jitsi-meet/issues/3514#issuecomment-427566880

**Comment by saghul** [comment]
There seems to be some bug in Xcode 10. Here is how to compile a release build to run on devices: https://github.com/jitsi/jitsi-meet/issues/3514#issuecomment-428176691

I'm closing this since there is nothing we can act on other than providing workarounds.
